### PR TITLE
Revert accidental change to logits activation

### PR DESCRIPTION
**Pull request by LukeWood**
At some point while we were trying to create a single decoder to use for MaskRCNN + RetinaNet the activation for the prediction decoder was changed to be softmax instead of sigmoid.  This has a few undesirable properties, but the most notable is that at times the model will make a prediction for a box with a logit of 0.000001 if the rest of the logits are all 0 or negative

This also removes the need for the extra class in RetinaNet.  We can just use len(class_ids) now.

### CODE REVIEW

**Review by jbischof — APPROVED** [review]
Could this be related to the high confidence false positives in Martin's bug dataset?

**Review by ianstenbit — APPROVED** [review]
Looks like we missed this in #1197

Nice catch Luke!
